kani-compiler/src/codegen_cprover_gotoc/context/goto_ctx.rs

```diff
@@ -99,6 +101,29 @@ impl<'tcx> GotocCtx<'tcx> {
     pub fn current_fn_mut(&mut self) -> &mut CurrentFnCtx<'tcx> {
         self.current_fn.as_mut().unwrap()
     }
+
+    /// Maps the goto-context "unsupported features" data into the
+    /// KaniMetadata "unsupported features" format.
+    ///
+    /// These are different because the KaniMetadata is a flat serializable list,
+    /// while we need a more richly structured HashMap in the goto context.
+    pub(crate) fn unsupported_metadata(&self) -> Vec<UnsupportedFeature> {
+        self.unsupported_constructs
+            .iter()
+            .map(|(construct, location)| UnsupportedFeature {
+                feature: construct.to_string(),
+                locations: location
+                    .iter()
+                    .map(|l| {
+                        (
+                            l.filename().unwrap_or_default(),
```

Comment:
This would be cleaner if `UnsupportedFeature` took an `Option` for filename and start line

Reply:
Only sort of? I think we actually have 0 instances of NOT having a location for our codegen_unimplemented calls. So in some sense, we're going from the too-general type (Location) to the more specific type.

I'll add a comment here.